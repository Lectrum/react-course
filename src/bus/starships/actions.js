// Types
import { types } from './types';

import { api } from '../../api';

export const starshipsActions = Object.freeze({
    // Sync
    fill: (payload) => {
        return {
            type: types.STARSHIPS_FILL,
            payload
        }
    },
    // Async
    fetchAsync: () => async (dispatch) => {
        dispatch({
            type: types.STARSHIPS_FETCH_ASYNC
        });
        
        const response = await api.starships.fetch();
        const { results } = await response.json();
        
        dispatch(starshipsActions.fill(results));
    }
});