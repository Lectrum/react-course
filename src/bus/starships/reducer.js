// Types
import { types } from './types';

const initialState = {
    data: []
};

export const starshipsReducer = ( state = initialState, { type, payload } ) => {
    switch (type) {
        case types.STARSHIPS_FILL:
            return { ...state, data: payload };
    
        default:
            return state;
    }
}