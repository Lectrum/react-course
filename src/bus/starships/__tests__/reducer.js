import { starshipsReducer } from '../reducer';
import { starshipsActions } from '../actions';

describe('starships reducer:', () => {
    test('should return initial state by default', () => {
        expect(starshipsReducer(void 0, {})).toMatchSnapshot();
    });
    test('should handle START_FETCHING action', () => {
        expect(starshipsReducer(void 0, starshipsActions.startFetching())).toMatchSnapshot();
    })
});