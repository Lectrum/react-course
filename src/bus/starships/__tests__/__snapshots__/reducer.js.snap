// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`starships reducer: should handle START_FETCHING action 1`] = `
Object {
  "data": Array [],
  "error": false,
  "isFetching": true,
}
`;

exports[`starships reducer: should return initial state by default 1`] = `
Object {
  "data": Array [],
  "error": false,
  "isFetching": false,
}
`;
